frontend/mobilityai/app/src/main/java/com/ai/mobility/mobilityai/MetaMotionService.java

```diff
@@ -1,224 +1,389 @@
 package com.ai.mobility.mobilityai;
```

Comment:
A lot of the code in this file is shared with `DeviceInfoActivity`. Does it make sense to pull it out into a utility class?

Reply:
All the stuff in DeviceInfoActivity is going to be removed. Most of it was copied and pasted into this one. I'll be cleaning that up next time since it'll require more testing to complete.